### PR TITLE
[Truffle] splat position; Thread#backtrace

### PR DESCRIPTION
**Pull request by pitr-ch**

### CODE REVIEW

**Comment by eregon** [comment]
I would call *arg in an argument list a `rest` argument instead of `splat`.
Why do you want the caller to perform the computation of the index, could Arity do it if we just pass required, optional and whether there is a rest argument?

**Comment by pitr-ch** [comment]
I think what you suggest is not enough because of post-splat arguments. The position can be smaller than required+optional. It could be pre, optional, splat?, post though.

**Comment by eregon** [comment]
How so? The frame will contain the arguments in order, so for instance optional will be either the given value or the default one.

File: truffle/src/main/java/org/jruby/truffle/nodes/arguments/CheckArityNode.java [hunk 1]
```diff
@@ -47,8 +47,8 @@ public void executeVoid(VirtualFrame frame) {
         final Object[] frameArguments = frame.getArguments();
         final int given;
         final RubyBasicObject keywordArguments;
-        
-        //TODO (MS): Check merge 
+
+        //TODO (MS): Check merge
```
Comment (eregon) [inline truffle/src/main/java/org/jruby/truffle/nodes/arguments/CheckArityNode.java]: Can you remove these unintentional changes?

Comment (pitr-ch) [inline truffle/src/main/java/org/jruby/truffle/nodes/arguments/CheckArityNode.java]: done, what's our policy on these changes, separate commits or don't? I feel these trailing spaces as malformed code.

Comment (eregon) [inline truffle/src/main/java/org/jruby/truffle/nodes/arguments/CheckArityNode.java]: ah sorry I misread the diff, I thought the diff added the extra spaces.
Removing neighboring style issues is fine.

**Comment by eregon** [comment]
Looks good, bonus point if you can rebase on master :)

**Pull request merged**